Makefile

```diff
@@ -4,14 +4,81 @@
 DEBUG  ?= 0
 OPENMP ?= 1
 MPI    ?= 0
-CXX     = g++
+CXX     = g++ 
 MKDIR_P = mkdir -p
 RM      = /bin/rm
 
 MAKE_CPPFLAGS = -D_GNU_SOURCE -Iinclude -I${GTEST_ROOT}/include
 MAKE_CXXFLAGS = -Wall -std=c++11
 MAKE_LDFLAGS  = -L. -L${GTEST_ROOT} -lgtest
 
+
+
+ifeq ($(PE_ENV), CRAY)
```

Comment:
If we want this tool to be generic, I think we should not rely to Cray's `PE_ENV`. Instead, we should set a variable at the makefile level. I'd see the following as a use case:

```
make toolchain=gnu
make toolchain=cray
make toolchain=pgi
```

Then we can check inside the `Makefile` for this variable. The default should `gnu`.